core/miner/impl/miner_impl.cpp

```diff
@@ -78,7 +85,39 @@ namespace fc::miner {
             api,
             deadline_info.period_start % kWPoStProvingPeriod,
             kMaxSectorExpirationExtension - 2 * kWPoStProvingPeriod);
-
+    std::shared_ptr<FeeConfig> fee_config = std::make_shared<FeeConfig>();
+    fee_config->max_precommit_batch_gas_fee.base = {
+        0};  // TODO: config loading;
+    fee_config->max_precommit_batch_gas_fee.per_sector =
+        static_cast<TokenAmount>(0.02 * 10e18);
+    fee_config->max_precommit_gas_fee = static_cast<TokenAmount>(0.025 * 10e18);
+    std::shared_ptr<PreCommitBatcher> precommit_batcher =
+        std::make_shared<PreCommitBatcherImpl>(
+            std::chrono::seconds(60),
+            api,
+            miner_address,
+            scheduler,
+            [=](MinerInfo miner_info,
+                TokenAmount deposit,
+                TokenAmount good_funds) -> outcome::result<Address> {
+              Address minimal_balanced_address;
+              auto finder = miner_info.control.end();
+              for (auto address = miner_info.control.begin();
+                   address != miner_info.control.end();
+                   ++address) {
+                if (api->WalletBalance(*address).value() >= good_funds
```

Comment:
`WalletBalance` is rpc, save result for `*address` to local variable, and copy that to `min/max`

Reply:
fixed